### PR TITLE
Allow schedule to be updated for embedded tasks (med tracking)

### PR DESCRIPTION
**Pull request by syoung-smallwisdom**

### CODE REVIEW

File: BridgeAppSDK/SBAActivityTableViewController.swift [hunk 1]
```diff
@@ -168,8 +175,28 @@ public class SBAActivityTableViewController: UITableViewController, SBASharedInf
...
         
+        // Add any additional schedules
+        var scheduledActivities = [schedule]
+        
+        // Look at top-level steps for a subtask that might have it's own schedule
+        if let navTask = taskViewController.task as? SBANavigableOrderedTask {
+            for step in navTask.steps {
+                if let subtaskStep = step as? SBASubtaskStep, let taskId = subtaskStep.taskIdentifier,
+                   let subschedule = scheduledActivityForTaskIdentifier(taskId) where !subschedule.isCompleted {
+                    // If schedule is found then set it's start/stop time and add to list to update
```
Comment (Erin-Mounts) [inline BridgeAppSDK/SBAActivityTableViewController.swift]: it's -> its

File: BridgeAppSDK/SBABridgeTask+Dictionary.swift [hunk 2]
```diff
@@ -68,7 +68,7 @@ extension NSDictionary: SBABridgeTask {
     }
 
     public var schemaRevision: NSNumber! {
-        return self["schemaRevision"] as? NSNumber ?? NSNumber(integer:1)
+        return self["schemaRevision"] as? NSNumber ?? NSNumber(integer:0)
```
Comment (Erin-Mounts) [inline BridgeAppSDK/SBABridgeTask+Dictionary.swift]: ?? default schema revision is 1--you seem to have changed this throughout BridgeAppSDK...?

File: BridgeAppSDK/SBASubtaskStep.swift [hunk 3]
```diff
@@ -38,6 +38,10 @@ import ResearchKit
  */
 public class SBASubtaskStep: ORKStep {
     
+    public var taskIdentifier: String?
+    public var schemaIdentifier: String?
+    public var schemaRevision: Int = 0
```
Comment (Erin-Mounts) [inline BridgeAppSDK/SBASubtaskStep.swift]: 1?

File: BridgeAppSDK/SBATrackedDataObjectCollection.m [hunk 4]
```diff
@@ -54,7 +54,7 @@ - (NSString *)schemaIdentifier {
 
 - (NSNumber *)schemaRevision {
     if (_schemaRevision == nil) {
-        _schemaRevision = @(1);
+        _schemaRevision = @(0);
```
Comment (Erin-Mounts) [inline BridgeAppSDK/SBATrackedDataObjectCollection.m]: 1?

**Comment by Erin-Mounts** [comment]
one nit about syntax in a comment, and a broader question about setting the default schemaRevision to 0 throughout--otherwise looks good

**Comment by syoung-smallwisdom** [comment]
Updated commit to remove the schemaRevision references until next week when looking at archiving.

**Comment by Erin-Mounts** [comment]
looks good

**Pull request merged**